commit 53d19f3961a393ccf6fbaba99c40559928670a54
Author: zhongshanwu <zhongshanwu@goodix.com>
Date:   Mon Aug 14 16:09:56 2017 +0800

    ubunt modify1

diff --git a/gt9xx.c b/gt9xx.c
index eaa2693..39d9221 100644
--- a/gt9xx.c
+++ b/gt9xx.c
@@ -1944,7 +1944,7 @@ static s8 gtp_request_input_dev(struct goodix_ts_data *ts)
 
     ts->input_dev->evbit[0] = BIT_MASK(EV_SYN) | BIT_MASK(EV_KEY) | BIT_MASK(EV_ABS) ;
 #if GTP_ICS_SLOT_REPORT
-    input_mt_init_slots(ts->input_dev, 16, 0);     // in case of "out of memory"
+    input_mt_init_slots(ts->input_dev, 16, INPUT_MT_DIRECT | INPUT_MT_DROP_UNUSED);     // in case of "out of memory"
 #else
     ts->input_dev->keybit[BIT_WORD(BTN_TOUCH)] = BIT_MASK(BTN_TOUCH);
 #endif
@@ -1970,16 +1970,19 @@ static s8 gtp_request_input_dev(struct goodix_ts_data *ts)
     input_set_abs_params(ts->input_dev, ABS_MT_POSITION_Y, 0, ts->abs_y_max, 0, 0);
     input_set_abs_params(ts->input_dev, ABS_MT_WIDTH_MAJOR, minv, 255, 0, 0);
     input_set_abs_params(ts->input_dev, ABS_MT_TOUCH_MAJOR, minv, 255, 0, 0);
-    input_set_abs_params(ts->input_dev, ABS_MT_TRACKING_ID, 0, 255, 0, 0);
+    //input_set_abs_params(ts->input_dev, ABS_MT_TRACKING_ID, 0, 255, 0, 0);// 170814 delete tracking id
 
     //ts->input_dev->name = goodix_ts_name;
     //ts->input_dev->name = "hideep-ist771x";
     ts->input_dev->name = "lenovo-smartpad_ts";
-    ts->input_dev->phys = goodix_input_phys;
+    ts->input_dev->phys = "input/ts";//170814 modify phys
     ts->input_dev->id.bustype = BUS_I2C;
-    ts->input_dev->id.vendor = 0xDEAD;
+    ts->input_dev->id.vendor = 0x0416;//170814 modify vendor
     ts->input_dev->id.product = 0xBEEF;
     ts->input_dev->id.version = 10427;
+
+    /* 170814 additional set drvdata for input device */
+    input_set_drvdata(ts->input_dev, ts);
     
     ret = input_register_device(ts->input_dev);
     if (ret)
